src/mlpack/bindings/cli/CMakeLists.txt

```diff
@@ -50,11 +50,20 @@ if (BUILD_CLI_EXECUTABLES)
   add_executable(mlpack_${name}
     ${name}_main.cpp
   )
+
+  if(NOT BUILD_SHARED_LIBS)
```

Comment:
Maybe we should rename `BUILD_SHARED_LIBS` to `BUILD_SHARED_OBJECTS` (or add another configuration flag)?  Technically these command-line programs are not libraries.

Reply:
Are you sure, this variable only to check if mlpack and other dependencies are to build as `.a` therefore allowing to build in static. Let me know what do you think

Reply:
I suppose... we should point out somewhere that if `BUILD_SHARED_LIBS` is `OFF`, then even the command-line programs will be linked statically.  Either that, or just add another option `BUILD_SHARED_PROGRAMS` or something.  I think either way is fine.

Reply:
I have added a comment in the main `CMakeLists.txt` Let me know if this makes sense.